client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx

```diff
@@ -84,7 +91,15 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 		setSiteDescription( tagline );
 		setSiteTitle( siteTitle );
 		setSiteAccentColor( accentColor.hex );
-		setPaidSubscribers( paidSubscribers );
+		setPaidSubscribers( paidSubscribers ); // @TODO delete this?
+
+		if ( paidSubscribers ) {
+			console.log( 'Setting paid subscriber status YES 💚' ); // eslint-disable-line no-console
+			setGoals( Onboard.SiteGoal.PaidSubscribers );
+		} else {
+			console.log( 'Setting paid subscriber status NO 🍎' ); // eslint-disable-line no-console
+			resetGoals();
```

Comment:
Fires currently even if goals were never set. But we also need to ensure we do clear them if user comes back to this step and unselects, and resubmits.